### ISSUE TITLE
Does not wake-up after standby/freeze sometimes after recent update (AMD)

### ISSUE THREAD

**Issue by rugk** [fedora-silverblue/issue-tracker#495]
**Describe the bug**
After updating from 38.20230920.0 to 38.20230927.0 (I am almost sure it has happened since that update), my laptop often seems to be stuck with a black screen (/in standby) when I try to wake it up from standby. It is, as if it does not react on the press of the power-button. Then, I often need to force-reboot it (via the power button), which is unfortunate.
Has happened like I dunno two times maybe now? 

Also, possibly related, it froze one time just now when I had an external display connected via HDMI (no adapter, HDMI port at laptop) and USB-C for power delivery at the same time, the power plug may have disconnected and caused this issue). I also had to force reboot.

Never had that before, it is quite out of the sudden that these issues happen…

**To Reproduce**
Try the versions. Likely component- or device-specific.

**Expected behavior**
Not hang.

**Screenshots**
N/A

**OS version:**
```
State: idle
AutomaticUpdates: stage; rpm-ostreed-automatic.timer: last run 1h 38min ago
BootedDeployment:
● fedora:fedora/38/x86_64/silverblue
                  Version: 38.20230927.0 (2023-09-27T02:13:44Z)
               BaseCommit: 84a1d651086ddda05cdcd52167919f11688f7c9283687e97118c1f7336284a66
             GPGSignature: Valid signature by 6A51BBABBA3D5467B6171221809A8D7CEB10B464
          LayeredPackages: adb blivet-gui btop dconf-editor firewall-config git git-credential-libsecret *** nvme-cli openssl
                           pipewire-codec-aptx podman-compose rpmfusion-free-release simple-scan smartmontools sushi
                           tldr zsh         
```

**Additional context**
```
● fedora:fedora/38/x86_64/silverblue (index: 1)
                  Version: 38.20230927.0 (2023-09-27T02:13:44Z)
               BaseCommit: 84a1d651086ddda05cdcd52167919f11688f7c9283687e97118c1f7336284a66
                           ├─ repo-0 (2023-04-13T20:37:10Z)
                           ├─ repo-1 (2023-09-27T01:40:03Z)
                           └─ repo-2 (2023-09-27T01:43:59Z)
                   Commit: dea94825e0db2170bba66011a2eac7492fef2df1a7af87d61160d325f4bd8e9a
                           ├─ fedora (2023-04-13T20:37:10Z)
                           ├─ fedora-cisco-openh264 (2023-03-14T10:56:46Z)
                           ├─ fedora-modular (2023-04-13T20:30:47Z)
                           ├─ rpmfusion-free (2023-04-14T11:37:11Z)
                           ├─ rpmfusion-free-updates (2023-09-26T11:20:14Z)
                           ├─ updates (2023-09-27T02:38:11Z)
                           ├─ updates-archive (2023-09-26T01:48:51Z)
                           └─ updates-modular (2023-09-16T00:25:03Z)
                StateRoot: fedora
             GPGSignature: 1 signature
                           Signature made Mi 27 Sep 2023 04:13:48 CEST using RSA key ID 809A8D7CEB10B464
                           Good signature from "Fedora <fedora-38-primary@fedoraproject.org>"
          LayeredPackages: adb blivet-gui btop dconf-editor firewall-config git git-credential-libsecret *** nvme-cli openssl
                           pipewire-codec-aptx podman-compose rpmfusion-free-release simple-scan smartmontools sushi
                           tldr zsh                   

  fedora:fedora/38/x86_64/silverblue (index: 2)
                  Version: 38.20230920.0 (2023-09-20T00:52:52Z)
               BaseCommit: 5fbec9c2714ffb9ee8a73093fbb101b707e59164d689b7c0057c7d8dadc23712
                           ├─ repo-0 (2023-04-13T20:37:10Z)
                           ├─ repo-1 (2023-09-20T00:16:59Z)
                           └─ repo-2 (2023-09-20T00:20:15Z)
                   Commit: 9eecd46a39435af7169b008c2a00b1fb253e388cdc62d5acafeeb0bb2e9dd1b6
                           ├─ fedora (2023-04-13T20:37:10Z)
                           ├─ fedora-cisco-openh264 (2023-03-14T10:56:46Z)
                           ├─ fedora-modular (2023-04-13T20:30:47Z)
                           ├─ rpmfusion-free (2023-04-14T11:37:11Z)
                           ├─ rpmfusion-free-updates (2023-09-14T21:01:39Z)
                           ├─ updates (2023-09-20T01:22:56Z)
                           ├─ updates-archive (2023-09-20T02:19:42Z)
                           └─ updates-modular (2023-09-16T00:25:03Z)
                StateRoot: fedora
             GPGSignature: 1 signature
                           Signature made Mi 20 Sep 2023 02:52:57 CEST using RSA key ID 809A8D7CEB10B464
                           Good signature from "Fedora <fedora-38-primary@fedoraproject.org>"
          LayeredPackages: adb blivet-gui btop dconf-editor firewall-config git git-credential-libsecret *** nvme-cli openssl
                           pipewire-codec-aptx podman-compose rpmfusion-free-release simple-scan smartmontools sushi
                           tldr zsh                        
```

As there were some firmware and kernel updates, I suspect this may be related to the software/updates:

```
$ rpm-ostree db diff 9eecd46a39435af7169b008c2a00b1fb253e388cdc62d5acafeeb0bb2e9dd1b6 dea94825e0db2170bba66011a2eac7492fef2df1a7af87d61160d325f4bd8e9a
ostree diff commit from: 9eecd46a39435af7169b008c2a00b1fb253e388cdc62d5acafeeb0bb2e9dd1b6
ostree diff commit to:   dea94825e0db2170bba66011a2eac7492fef2df1a7af87d61160d325f4bd8e9a
Upgraded:
  amd-gpu-firmware 20230804-153.fc38 -> 20230919-1.fc38
  atheros-firmware 20230804-153.fc38 -> 20230919-1.fc38
  brcmfmac-firmware 20230804-153.fc38 -> 20230919-1.fc38
  container-selinux 2:2.221.1-1.fc38 -> 2:2.222.0-1.fc38
  cups 1:2.4.6-4.fc38 -> 1:2.4.7-1.fc38
  cups-client 1:2.4.6-4.fc38 -> 1:2.4.7-1.fc38
  cups-filesystem 1:2.4.6-4.fc38 -> 1:2.4.7-1.fc38
  cups-ipptool 1:2.4.6-4.fc38 -> 1:2.4.7-1.fc38
  cups-libs 1:2.4.6-4.fc38 -> 1:2.4.7-1.fc38
  gstreamer1 1.22.5-1.fc38 -> 1.22.6-1.fc38
  intel-gpu-firmware 20230804-153.fc38 -> 20230919-1.fc38
  iwlegacy-firmware 20230804-153.fc38 -> 20230919-1.fc38
  iwlwifi-dvm-firmware 20230804-153.fc38 -> 20230919-1.fc38
  iwlwifi-mvm-firmware 20230804-153.fc38 -> 20230919-1.fc38
  kernel 6.4.15-200.fc38 -> 6.5.5-200.fc38
  kernel-core 6.4.15-200.fc38 -> 6.5.5-200.fc38
  kernel-modules 6.4.15-200.fc38 -> 6.5.5-200.fc38
  kernel-modules-core 6.4.15-200.fc38 -> 6.5.5-200.fc38
  kernel-modules-extra 6.4.15-200.fc38 -> 6.5.5-200.fc38
  langtable 0.0.63-1.fc38 -> 0.0.64-1.fc38
  libertas-firmware 20230804-153.fc38 -> 20230919-1.fc38
  libppd 1:2.0~rc2-3.fc38 -> 1:2.0~rc2-4.fc38
  libwebp 1.3.1-3.fc38 -> 1.3.2-1.fc38
  linux-firmware 20230804-153.fc38 -> 20230919-1.fc38
  linux-firmware-whence 20230804-153.fc38 -> 20230919-1.fc38
  mesa-dri-drivers 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-filesystem 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-libEGL 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-libGL 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-libgbm 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-libglapi 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-libxatracker 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-va-drivers 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mesa-vulkan-drivers 23.1.7-1.fc38 -> 23.1.8-1.fc38
  mt7xxx-firmware 20230804-153.fc38 -> 20230919-1.fc38
  nvidia-gpu-firmware 20230804-153.fc38 -> 20230919-1.fc38
  pipewire-codec-aptx 0.3.79-1.fc38 -> 0.3.80-1.fc38
  python3-langtable 0.0.63-1.fc38 -> 0.0.64-1.fc38
  realtek-firmware 20230804-153.fc38 -> 20230919-1.fc38
  vim-data 2:9.0.1872-1.fc38 -> 2:9.0.1927-1.fc38
  vim-minimal 2:9.0.1872-1.fc38 -> 2:9.0.1927-1.fc38
  zip 3.0-36.fc38 -> 3.0-37.fc38
```

I have pinned the old deployment for now.
And no sorry did not have time to install Fedora Workstation to reproduce it there… :see_no_evil: 

## device

Lenovo IdeaPad 5 15ARE05
CPU: AMD Ryzen 7 4800U with Radeon Graphics
GPU: none (integrated)

Full `lshw` output: https://gist.github.com/rugk/d6517f7309a9ceedadaf929daf3e00be

---

Cross-posted [to AskFedora](https://discussion.fedoraproject.org/t/does-not-wake-up-after-standby-freeze-sometimes-after-recent-update-amd/91109?u=rugk).

**Comment by travier** [comment]
This is likely a kernel regression:

```
kernel 6.4.15-200.fc38 -> 6.5.5-200.fc38
```

I would recommend downgrading the kernel to confirm and then filling a bug for the kernel.

**Comment by chsjr1996** [comment]
Same here, AMD Graphics too and after update to kernel 6.5.5 the system randomly freezes. Then, I'm rollback to previous deployment (kernel 6.4.15-200) and all works again.

---

**Some important links:**

https://bodhi.fedoraproject.org/updates/FEDORA-2023-4c8291ba6a
https://gitlab.freedesktop.org/drm/amd/-/issues/2830

**Comment by rugk** [comment]
> then filling a bug for the kernel

So is this still needed? Seeing the previous links the issue seems to be known (although I saw no kernel bugzilla link).

**Comment by rugk** [comment]
Tracked in https://bugzilla.redhat.com/show_bug.cgi?id=2240859 and apparently fixed in kernel v6.5.6.
